automates/model_analysis/identification_algorithms.py

```diff
@@ -0,0 +1,251 @@
+        tree.call.id_check = all(id_check_list)
+        return gm.ResultsInternal(p=gm.Probability(sumset=gm.ts(list(set(v) - set(y) - set(x)), topo), product=True, \
+                                                   children=product_list), tree=tree)
+    else:
+        s_single = s[0]
+
+        # Line 5
+        cc = gm.c_components(g, topo)
+        if cc[0] == v:
+            tree.call.s = cc[0]
+            tree.call.line = 5
+            tree.call.id_check = False
+            tree.root = deepcopy(p)
+            return gm.ResultsInternal(p=p, tree=tree)
+
+        # Line 6
+        if s_single in cc:
+            tree.call.line = 6
+            tree.call.s = s_single
+            s_single_length = len(s_single)
+            product_list = []
+            p_prod = gm.Probability()
+            for node in s_single:
+                node_topo_index = v.index(node)
+                cond_set = v[0:node_topo_index]
+                if p.product:
+                    p_prod = gm.parse_joint(p, node, cond_set, v, topo)
+                else:
+                    p_prod = deepcopy(p)
+                    p_prod.var = [node]
+                    p_prod.cond = cond_set
+                product_list.append(p_prod)
+            product_list.reverse()
+            if s_single_length > 1:
+                prob_new = gm.Probability(sumset=gm.ts(set(s_single) - set(y), topo), product=True, children= \
+                    product_list)
+                tree.root = prob_new
+                tree.call.id_check = True
+                return gm.ResultsInternal(p=prob_new, tree=tree)
+            if p_prod.product or p_prod.fraction:
+                p_prod.sumset = gm.ts(list(set(p_prod.sumset) | (set(s_single) - set(y))), topo)
+            else:
+                p_prod.var = gm.ts(set(p_prod.var) - (set(p_prod.sumset) | (set(s_single) - set(y))), topo)
+            tree.root = p_prod
+            tree.call.id_check = True
+            return gm.ResultsInternal(p=p_prod, tree=tree)
+
+        # Line 7
+        tree.call.s = s_single
+        s_prime = None
+        for component in cc:
+            if set(s_single).issubset(set(component)):
+                s_prime = component
+                break
+        tree.call.line = 7
+        tree.call.s_prime = s_prime
+        s_prime_length = len(s_prime)
+        g_s_prime = g.subgraph(s_prime)
+        g_s_prime_obs = gm.observed_graph(g_s_prime)
+        product_list = []
+        p_prod = gm.Probability()
+        for node in s_prime:
+            node_topo_index = v.index(node)
+            cond_set = v[0:node_topo_index]
+            if p.product:
+                p_prod = gm.parse_joint(p, node, cond_set, v, topo)
+            else:
+                p_prod = deepcopy(p)
+                p_prod.var = [node]
+                p_prod.cond = cond_set
+            product_list.append(p_prod)
+        product_list.reverse()
+        x_new = gm.ts(set(x) & set(s_prime), topo)
+        if s_prime_length > 1:
```

Comment:
@aswinchester similar to my comment about the other conditional with large duplicated commands above, let's simply this statement to be the following:
```python
if s_prime_length > 1:
  alex_name_me_pls = gm.Probability(product=True, children=product_list)
else:
  alex_name_me_pls = product_list[0]

nxt = compute_ID(
    y, x_new, alex_name_me_pls, g_s_prime, g_s_prime_obs, s_prime, topo, gm.TreeNode()
)
```

Reply:
This has been completed